BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
```

Comment:
For financial values, `float` is not the correct choice. Use `decimal` for concepts like currencies, exchange rates, interest rates etc. It's a lot more precise than `float` or `double`. It's also less performant, but that is worth it when working within the financial domain, as we lower the risk of rounding errors.

In all other cases, stick with `double` instead of `float`. It's the default in C# and provides a higher precision at a negligible loss in performance.